app/services/create_cluster_stats_service.rb

```diff
@@ -36,17 +34,27 @@ def call
 
   private
 
+  def network_stat
+    @network_stat ||= ClusterStat.find_or_create_by(network: @network)
+  end
+
+  def epoch_duration
+    network_stat.epoch_duration ||
+      ClusterStats::UpdateEpochDurationService.new(@network).call.epoch_duration
+  end
+
   def epochs_annual_roi
     if last_epochs.count > 0
-      epoch_duration = (last_epochs.first.created_at - last_epochs.second.created_at)
-      return 0 unless epoch_duration.to_f > 0
-      epoch_average_roi = last_epochs.map do |epoch|
-        if epoch.total_rewards && epoch.total_active_stake
-          (epoch.total_rewards / epoch.total_active_stake.to_f) * 100
-        end
-      end.compact&.average.to_f
+      epoch_average_roi =
```

Comment:
Never break the lines this way. If you want to pull the variable name before block, it should be aligned as below.
```suggestion
      epoch_average_roi = last_epochs.map do |epoch|      
```